onnxruntime_extensions/tools/pre_post_processing/steps/nlp.py

```diff
@@ -0,0 +1,449 @@
+        assert input_type_str0 == "string"
+
+        assert len(input_shape_str0.split(",")) in [1, 2], "support either one (classification) or two sentences (QA)"
+        onnx_tokenizer_impl = "HfBertTokenizer" if len(input_shape_str0.split(",")) == 2 else "BertTokenizer"
+
+        def build_output_declare():
+            output_base = []
+            for out in self.output_names:
+                output_base.append(f"int64[{output_shape_str}] {out}")
+
+            return ",".join(output_base)
+
+        def get_tokenizer_ret():
+            if onnx_tokenizer_impl == "HfBertTokenizer":
+                return ",".join(self.output_names)
+
+            return f"ids,types,mask"
+
+        def build_output_imp():
+            if onnx_tokenizer_impl == "HfBertTokenizer":
+                return ""
+
+            ret_vars = get_tokenizer_ret().split(",")
+            ret_vars[1], ret_vars[2] = ret_vars[2], ret_vars[1]
+            output_str = [f"i64_0 = Constant <value = int64[1] {{0}}> ()"]
+
+            for i in range(0, len(self.output_names)):
+                output_str.append(f"{self.output_names[i]} = Unsqueeze({ret_vars[i]}, i64_0)")
+
+            return "\n".join(output_str)
+
+        def build_input_declare():
+            inputs = f"{input_type_str0}[{input_shape_str0}] {self.input_names[0]}"
+            return inputs
+
+        def build_tokenizer_call_arg():
+            call_args = f"{self.input_names[0]}"
+            return call_args
+
+        converter_graph = onnx.parser.parse_graph(
+            f"""\
+            tokenizer ({build_input_declare()}) 
+                => ({build_output_declare()})
+            {{
+                {get_tokenizer_ret()} =  {self.domain}.{onnx_tokenizer_impl} ({build_tokenizer_call_arg()})
+                {build_output_imp()}
+            }}
+            """
+        )
+
+        token_model_attr = []
+        bert_tokenizer = self.tokenizer
+        ordered_vocab = OrderedDict(sorted(bert_tokenizer.vocab.items(), key=lambda item: int(item[1])))
+        vocab = "\n".join(ordered_vocab.keys())
+        attrs = dict(vocab_file=vocab)
+
+        attrs["strip_accents"] = (
+            1
+            if "strip_accents" in bert_tokenizer.init_kwargs and bert_tokenizer.init_kwargs.get("strip_accents")
+            else 0
+        )
+        attrs["do_lower_case"] = 1 if hasattr(bert_tokenizer, "do_lower_case") and bert_tokenizer.do_lower_case else 0
+
+        for attr in attrs:
+            attr_value = onnx.helper.make_attribute(attr, attrs[attr])
+            token_model_attr.append(attr_value)
+
+        node_idx = next(i for i, v in enumerate(converter_graph.node) if v.op_type == onnx_tokenizer_impl)
+        converter_graph.node[node_idx].attribute.extend(token_model_attr)
+
+        return converter_graph
+
+
+class BertTokenizerQATask(Step):
+    def __init__(self, name: Optional[str] = None):
+        """
+        Brief:
+            Just copy input_ids for decoder
```

Comment:
Can this be more descriptive? This text becomes part of the documentation and should be meaningful to an external user who isn't familiar with all the internals of the class. i

Reply:
Sure. Done.